fw/sock.c

```diff
@@ -1005,6 +1005,9 @@ ss_tcp_data_ready(struct sock *sk)
 		return;
 	}
 
+	if (unlikely(SS_CONN_TYPE(sk) & Conn_Reset))
+		return;
```

Comment:
It is not a workaround, but a bugfix: In case of errors, we should exit the call chain and stop handling the involved TLS records, unrolled skb, and `sk->sk_receive_queue`. Otherwise, it's an undefined behavior. And yes, we didn't cover the `RST` case (not close) in that function.

Reply:
The new socket state used only in one function looks awkward.

Can we handle this with a real socket state? Maybe we can even avoid calling `sk_data_ready` by setting some socket state and/or doing partial close/reset?